pkg/iss/iss.go

```diff
@@ -687,27 +687,25 @@ func (iss *ISS) processCommitted() error {
 	// deliver the corresponding certificate and advance to the next sequence number.
 	for iss.commitLog[iss.nextDeliveredSN] != nil {
 
-		certData := iss.commitLog[iss.nextDeliveredSN].CertData
+		// Convenience variables.
+		logEntry := iss.commitLog[iss.nextDeliveredSN]
+		certData := logEntry.CertData
 
-		var cert availabilitypb.Cert
-		if len(certData) > 0 {
-			if err := proto.Unmarshal(certData, &cert); err != nil {
+		var cert *apbtypes.Cert
+		if logEntry.Aborted {
+			iss.LeaderPolicy.Suspect(iss.epoch.Nr(), logEntry.Suspect)
+			cert = &apbtypes.Cert{Type: &apbtypes.Cert_Mscs{Mscs: &mscpbtypes.Certs{}}}
+			// TODO: Using a dummy MSC cert here. A nil value would be more natural, as this data is not to be used,
+			//  but currently nil values are not supported in generated types. Use nil when nil support is implemented.
```

Comment:
If we have defined an empty cert as a slice with no values, would that not be better than having a nil value here?

Reply:
Fair enough, we do need to support nil values either way for the code generator, but I think it is probably a better practice to semantically create a default empty cert than to use nil (that we already know can cause certain issues later on). Let's leave the comment there for now